### PR TITLE
MODTEMPENG-80 Logging improvement

### PR DESCRIPTION
**Pull request by pavankumar181**
## Purpose
MODTEMPENG-80 (https://issues.folio.org/browse/MODTEMPENG-80) 
Adding loggers as part of logging improvements

## Approach
_By following the practices mentioned in https://wiki.folio.org/display/DD/Logging,
Added loggers wherever needed._

## Evidence 
image

### CODE REVIEW

File: src/main/java/org/folio/template/dao/TemplateDaoImpl.java [hunk 1]
```diff
@@ -35,40 +38,46 @@ public TemplateDaoImpl(PostgresClient postgresClient) {
...
     pgClient.getById(TEMPLATES_TABLE, id, Template.class, promise);
     return promise.future().map(Optional::ofNullable);
   }
 
   @Override
   public Future<String> addTemplate(Template template) {
+    LOG.debug("addTemplate:: Adding template to database by Template ID: {}", template.getId());
     Promise<String> promise = Promise.promise();
     pgClient.save(TEMPLATES_TABLE, template.getId(), template, promise);
     return promise.future();
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/dao/TemplateDaoImpl.java:67]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/dao/TemplateDaoImpl.java [hunk 2]
```diff
@@ -35,40 +38,46 @@ public TemplateDaoImpl(PostgresClient postgresClient) {
...
   }
 
   @Override
   public Future<Boolean> updateTemplate(Template template) {
+    LOG.debug("updateTemplate:: Updating template in database by Template ID: {}", template.getId());
     Promise<RowSet<Row>> promise = Promise.promise();
     pgClient.update(TEMPLATES_TABLE, template, template.getId(), promise);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/dao/TemplateDaoImpl.java:74]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/dao/TemplateDaoImpl.java [hunk 3]
```diff
@@ -35,40 +38,46 @@ public TemplateDaoImpl(PostgresClient postgresClient) {
...
     return promise.future().map(updateResult -> updateResult.rowCount() == 1);
   }
 
   @Override
   public Future<Boolean> deleteTemplate(String id) {
+    LOG.debug("deleteTemplate:: Deleting template from database by Template ID: {}", id);
     Promise<RowSet<Row>> promise = Promise.promise();
     pgClient.delete(TEMPLATES_TABLE, id, promise);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/dao/TemplateDaoImpl.java:82]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/service/TemplateServiceImpl.java [hunk 4]
```diff
@@ -55,16 +58,19 @@ public TemplateServiceImpl(Vertx vertx, Map<String, String> okapiHeaders) {
...
   @Override
   public Future<Optional<Template>> getTemplateById(String id) {
+    LOG.debug("getTemplateById:: Retrieving Template by ID : {}", id);
     return templateDao.getTemplateById(id);
   }
 
   @Override
   public Future<String> addTemplate(Template template) {
+    LOG.debug("addTemplate:: Adding Template with ID : {}", template.getId());
     validateTemplate(template);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/service/TemplateServiceImpl.java:74]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/service/TemplateServiceImpl.java [hunk 5]
```diff
@@ -55,16 +58,19 @@ public TemplateServiceImpl(Vertx vertx, Map<String, String> okapiHeaders) {
...
     if (template.getId() == null) {
       template.setId(UUID.randomUUID().toString());
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/service/TemplateServiceImpl.java:76]: Please add log for the condition block of code it has been missed many places. Please check all the places and follow the logging guide lines of this improvement work.

File: src/main/java/org/folio/template/service/TemplateServiceImpl.java [hunk 6]
```diff
@@ -55,16 +58,19 @@ public TemplateServiceImpl(Vertx vertx, Map<String, String> okapiHeaders) {
...
 
   public Future<List<Template>> getTemplates(String query, int offset, int limit) {
+    LOG.debug("getTemplates:: Retrieving Templates with query {}", query);
     return templateDao.getTemplates(query, offset, limit);
   }
 
   @Override
   public Future<Optional<Template>> getTemplateById(String id) {
+    LOG.debug("getTemplateById:: Retrieving Template by ID : {}", id);
     return templateDao.getTemplateById(id);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/service/TemplateServiceImpl.java:68]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/service/TemplateServiceImpl.java [hunk 7]
```diff
@@ -55,16 +58,19 @@ public TemplateServiceImpl(Vertx vertx, Map<String, String> okapiHeaders) {
   }
 
   public Future<List<Template>> getTemplates(String query, int offset, int limit) {
+    LOG.debug("getTemplates:: Retrieving Templates with query {}", query);
     return templateDao.getTemplates(query, offset, limit);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/service/TemplateServiceImpl.java:62]: Please add return info logs for this method before return statement.

File: src/main/java/org/folio/template/service/TemplateServiceImpl.java [hunk 8]
```diff
@@ -74,6 +80,7 @@ public Future<String> addTemplate(Template template) {
 
   @Override
   public Future<Boolean> updateTemplate(Template template) {
+    LOG.debug("updateTemplate:: Updating Template with ID : {}", template.getId());
     validateTemplate(template);
     return getTemplateById(template.getId())
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/template/service/TemplateServiceImpl.java:85]: Please add return info logs for this method before return statement.

**Review by gurleenkaurbp — CHANGES_REQUESTED** [review]
Please add return info logs for this method before return statement.
Please add condition block loggers, it has been missed at many places.
Please add the evidence screenshot of the logs where the logs are printed.

**Review by gurleenkaurbp — APPROVED** [review]

**Review by Arghya22 — APPROVED** [review]

**Pull request merged**